### DISCUSSION TITLE
Is a way to render a control to a bitmap without showing it on screen?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#6921] [Q&A]
Hi,

I can render a control to a bitmap as follow:

            var bmp = new RenderTargetBitmap(pxz, dpi);
            
            var win = ((Window)control);
            win.Show();
            bmp.Render(win);
            win.Close();

However, it seems that I must show the window otherwise there is no output.

Is there anyway to get a bitmap of a control or window without having to show on screen?

**Comment by wieslawsoltes** [comment]
win.ShowInTaskbar = false;
win.WindowState = WindowState.Minimized;

**Reply by kuiperzone** [reply]
Thank you!

**Comment by kuiperzone** [comment]
Actually, I think it needs this:

            win.ShowInTaskbar = false;
            win.WindowState = WindowState.Minimized;
            win.SystemDecorations = SystemDecorations.None;
